controllers/articles.controller.js

```diff
@@ -27,9 +27,26 @@ exports.patchArticleById = (req, res, next) => {
 };
 
 exports.getArticles = (req, res, next) => {
-  selectArticles().then((articles) => {
-    res.status(200).send({ articles });
-  });
+  if (Object.keys(req.query).length > 0) {
+    if (
+      req.query.hasOwnProperty('sort_by') ||
+      req.query.hasOwnProperty('order_by') ||
+      req.query.hasOwnProperty('topic')
+    ) {
+    } else {
+      return Promise.reject({ status: 400, msg: 'Misspelt query!' }).catch(
+        next
+      );
+    }
+  }
```

Comment:
Lines 30 - 41 should really be happening in the model as it is validating the query and throwing errors based on it - then if you throw the error in the model it will be caught by the catch on line 49 and the middleware will all function as expected

Reply:
Not sure what this one means but I have put all error handling and data management into the model